### PR TITLE
show test failing with hasChangedAttributes

### PR DESCRIPTION
**Pull request by GuillaumeCisco**

### CODE REVIEW

**Comment by patocallaghan** [comment]
@GuillaumeCisco Apologies I was on vacation for the last week. I wonder if you could put your potential dummy fix in this PR too and maybe update the description? If anything to generate discussion about how to properly fix it. I don't know a whole lot about the internals of RecordData but we can ping the relevant folks to get them to look

File: tests/unit/fragment_array_property_test.js [hunk 1]
```diff
@@ -140,20 +140,25 @@ module('unit - `MF.fragmentArray` property', function(hooks) {
         region: 'Crownlands',
         country: 'Westeros'
       });
-      assert.deepEqual(person.changedAttributes().addresses, undefined, 'a frehsly pushed object has no changes');
+      assert.deepEqual(person.changedAttributes().addresses, undefined, 'a freshly pushed object has no changes');
+      assert.ok(!person._internalModel.hasChangedAttributes(), 'a freshly pushed object has no changed attributes');
```
Comment (patocallaghan) [inline tests/unit/fragment_array_property_test.js:144]: Does `hasDirtyAttributes` give us the same results as `person._internalModel.hasChangedAttributes()`?

**Comment by GuillaumeCisco** [comment]
> @GuillaumeCisco Apologies I was on vacation for the last week. I wonder if you could put your potential dummy fix in this PR too and maybe update the description? If anything to generate discussion about how to properly fix it. I don't know a whole lot about the internals of RecordData but we can ping the relevant folks to get them to look

The PR with the fix can be found on my fork here: https://github.com/GuillaumeCisco/ember-data-model-fragments/pull/1
In fact, it looks like to me `hasChangedAttributes` has never been correctly handled... It looks like a copy/paste from the original ember-data.
I don't know neither about how works RecordData so I cannot provide a smarter fix, help is welcome here :)

File: tests/unit/fragment_array_property_test.js (hunk 1 above)
Comment (GuillaumeCisco) [reply tests/unit/fragment_array_property_test.js:144]: I don't even know. I think it should by reading the [documentation](https://guides.emberjs.com/v3.22.0/models/creating-updating-and-deleting-records/)
But `hasChangedAttributes` is an internal method, and this is the one we need to update for fixing the bug in my case.
As `hasChangedAttributes` will be called as I developed it in earlier post.

**Comment by patocallaghan** [comment]
@GuillaumeCisco I think we can close this PR. It appears that the fix in https://github.com/lytics/ember-data-model-fragments/pull/381 makes the test you created here pass. They also include a fix with their test.

**Comment by patocallaghan** [comment]
The fix from #381 has been released in [v5.0.0-beta.1](https://github.com/adopted-ember-addons/ember-data-model-fragments/releases/tag/v5.0.0-beta.1)

**Pull request closed without merging**